lib/redmine_git_hosting/patches/repositories_controller_patch.rb

```diff
@@ -141,12 +141,18 @@ def diff_with_options
                           Digest::MD5.hexdigest("#{@path}-#{@rev}-#{@rev_to}-#{@diff_type}-#{current_language}")
           unless read_fragment(@cache_key)
             @diff = @repository.diff(@path, @rev, @rev_to)
+            unless @diff
+              show_error_not_found
+              return
+            end
+
             show_error_not_found unless @diff
```

Comment:
There is a duplication here. I think just adding `return` in front of  `show_error_not_found unless @diff` should work.

Reply:
Hi Nicolas, you are right, this was duplicated, I removed it with my last commit.